migrator/src/main/java/com/google/phonenumbers/migrator/MetadataReader.java

```diff
@@ -0,0 +1,74 @@
+package com.google.phonenumbers.migrator;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.i18n.phonenumbers.metadata.model.RangesTableSchema;
+import com.google.i18n.phonenumbers.metadata.table.CsvTable;
+import com.google.i18n.phonenumbers.metadata.table.RangeKey;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Optional;
+import java.util.Scanner;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * A class used to represent a given metadata zip file. Each {@link MetadataReader} instance
+ * represents a zip file containing Csv Tables that can be imported based on specified
+ * region codes without the need to extract the whole zip file.
+ */
+public final class MetadataReader {
+  private final ZipFile metadataZipFile;
+
+  public MetadataReader(String fileLocation) throws IOException {
```

Comment:
[Code Structure] When a class is final, there's often a slightly better alternative to public constructors; a "static factory method". This sounds fancy, but it really just means "a static method that calls the constructor".

What's the point? I hear you ask (well, I hope that's what you're doing) ... good question!
See https://effectivejava.googleplex.com/third-edition#page=26 which does a much better job at explaining that than me. Here could can also move the creation of the ZipFile into the factory method and just have the constructor check for non-null and assign the field (see com.google.common.base.Preconditions.checkNotNull).